Pod/Classes/SEGAmplitudeIntegration.m

```diff
@@ -159,11 +159,21 @@ - (void)screen:(SEGScreenPayload *)payload
 
 - (void)group:(SEGGroupPayload *)payload
 {
-    NSString *groupId = payload.groupId;
-    if (groupId) {
-        [self.amplitude setGroup:@"[Segment] Group" groupName:groupId];
-        SEGLog(@"[Amplitude setGroup:@'[Segment] Group' groupName:%@]", groupId);
+    NSString *groupTypeTrait = self.settings[@"groupTypeTrait"];
```

Comment:
This block:

```
    NSString *groupTypeTrait = self.settings[@"groupTypeTrait"];
    NSString *groupTypeValue = self.settings[@"groupTypeValue"];
    NSString *groupName;
    NSString *groupValue;

    if (payload.traits[groupTypeTrait] && payload.traits[groupTypeValue]) {
        groupName = payload.traits[groupTypeTrait];
        groupValue = payload.traits[groupTypeValue];
    } else {
        groupName = payload.traits[@"name"] ?: @"[Segment] Group";
        groupValue = payload.groupId;
    }
```

Is likely more readable and less error prone as:


```
    NSString *groupTypeTrait = self.settings[@"groupTypeTrait"];
    NSString *groupTypeValue = self.settings[@"groupTypeValue"];
    NSString *groupName = payload.traits[groupTypeTrait];
    NSString *groupValue = payload.traits[groupTypeValue];

    if (!groupName || !groupValue) {
        groupName = payload.traits[@"name"] ?: @"[Segment] Group";
        groupValue = payload.groupId;
    }
```

Reply:
https://github.com/segment-integrations/analytics-ios-integration-amplitude/compare/fix/group?expand=1